src/adaptations/device-layer/LwIP/WarmSupport.cpp

```diff
@@ -304,18 +304,36 @@ PlatformResult AddRemoveThreadAddress(InterfaceType inInterfaceType, const Inet:
 {
     otError otErr;
     otNetifAddress otAddress;
+    otBorderRouterConfig brConfig;
 
     memset(&otAddress, 0, sizeof(otAddress));
+
+    ThreadStackMgrImpl().LockThreadStack();
+
     otAddress.mAddress = ToOpenThreadIP6Address(inAddress);
     otAddress.mPrefixLength = 64;
     otAddress.mValid = true;
     otAddress.mPreferred = true;
 
-    ThreadStackMgrImpl().LockThreadStack();
+    brConfig.mConfigure = false;
+    brConfig.mDefaultRoute = false;
+    brConfig.mDhcp = false;
+    brConfig.mOnMesh = true;
+    brConfig.mPreference = 0;
+    brConfig.mPreferred = true;
+    brConfig.mPrefix.mPrefix = otAddress.mAddress;
+    brConfig.mPrefix.mLength = 64;
+    brConfig.mRloc16 = otThreadGetRloc16(ThreadStackMgrImpl().OTInstance());
+    brConfig.mSlaac = false;
+    brConfig.mStable = true;
 
     if (inAdd)
     {
         otErr = otIp6AddUnicastAddress(ThreadStackMgrImpl().OTInstance(), &otAddress);
+        if (otErr == OT_ERROR_NONE)
+        {
+            otErr = otBorderRouterAddOnMeshPrefix(ThreadStackMgrImpl().OTInstance(), &brConfig);
```

Comment:
This doesn't follow the WARM spec.  In particular, a device that isn't built to be a border router, or isn't currently operating as a border router, shouldn't be adding an on-mesh prefix.  This behavior is supposed to happen in StartStopThreadAdvertisement().

Reply:
In wpantund, https://github.com/openthread/wpantund/blob/1f656564af7b0efb137bbada4c6c9ae7d99adf26/src/wpantund/NCPInstanceBase-Addresses.cpp#L635-L667

When a unicast address is added to interface, a corresponding OnMeshPrefix is also added to the interface. Should we follow this behavior?

Also, when trying to add a /48 OnMeshPrefix in StartStopThreadAdvertisement, which is triggered at https://github.com/openweave/openweave-core/blob/2bcd0349d0fd11f73fa715405308651746871511/src/warm/WarmCore.cpp#L835-L842, thread will drop the packets, but a /64 on mesh prefix works fine.